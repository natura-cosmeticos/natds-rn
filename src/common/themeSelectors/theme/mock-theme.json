{
  "avatarSizes": {
    "tiny": {
      "size": 24,
      "fontSize": ".625rem"
    },
    "small": {
      "size": 32,
      "fontSize": ".875rem"
    },
    "standard": {
      "size": 40,
      "fontSize": "1rem"
    },
    "large": {
      "size": 48,
      "fontSize": "1rem"
    },
    "huge": {
      "size": 80,
      "fontSize": "1.5rem"
    }
  },
  "buttonSizes": {
    "small": {
      "paddingTop": 8,
      "paddingRight": 8,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "minHeight": 32
    },
    "medium": {
      "paddingTop": 12,
      "paddingRight": 16,
      "paddingBottom": 12,
      "paddingLeft": 16,
      "minHeight": 40
    },
    "large": {
      "paddingTop": 16,
      "paddingRight": 16,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "minHeight": 48
    }
  },
  "fontSize": {
    "h1": {
      "fontSize": "6rem",
      "fontWeight": 300
    },
    "h2": {
      "fontSize": "3.75rem",
      "fontWeight": 300
    },
    "h3": {
      "fontSize": "3rem",
      "fontWeight": 400
    },
    "h4": {
      "fontSize": "2.125rem",
      "fontWeight": 400
    },
    "h5": {
      "fontSize": "1.5rem",
      "fontWeight": 400
    },
    "h6": {
      "fontSize": "1.25rem",
      "fontWeight": 600
    },
    "subtitle1": {
      "fontSize": "1rem",
      "fontWeight": 400
    },
    "subtitle2": {
      "fontSize": "0.875rem",
      "fontWeight": 500
    },
    "body1": {
      "fontSize": "1rem",
      "fontWeight": 400
    },
    "body2": {
      "fontSize": "0.875rem",
      "fontWeight": 400
    },
    "button": {
      "fontSize": "0.875rem",
      "fontWeight": 600
    },
    "caption": {
      "fontSize": "0.75rem",
      "fontWeight": 400
    },
    "overline": {
      "fontSize": "0.75rem",
      "fontWeight": 400
    }
  },
  "iconSizes": {
    "micro": 16,
    "tiny": 24,
    "small": 32,
    "standard": 48,
    "medium": 64,
    "large": 72,
    "largex": 96,
    "largexx": 128,
    "huge": 144,
    "hugex": 192,
    "hugexx": 256
  },
  "opacity": {
    "12": 0.12,
    "16": 0.16,
    "24": 0.24,
    "32": 0.32,
    "48": 0.48,
    "56": 0.56,
    "64": 0.64,
    "80": 0.8,
    "00": 0,
    "04": 0.04,
    "08": 0.08
  },
  "overlay": {
    "solidBlack": 0.56
  },
  "radius": {
    "none": 0,
    "circle": "50%",
    "small": 2,
    "medium": 4,
    "large": 8
  },
  "sizes": {
    "none": 0,
    "micro": 4,
    "tiny": 8,
    "small": 16,
    "standard": 24,
    "semi": 32,
    "semix": 40,
    "medium": 48,
    "mediumx": 56,
    "large": 64,
    "largex": 72,
    "largexx": 80,
    "largexxx": 88,
    "huge": 96,
    "hugex": 128,
    "hugexx": 144,
    "hugexxx": 192,
    "veryhuge": 256
  },
  "spacing": {
    "spacing": 8,
    "spacingNone": 0,
    "spacingMicro": 4,
    "spacingTiny": 8,
    "spacingSmall": 16,
    "spacingStandard": 24,
    "spacingSemi": 32,
    "spacingLarge": 48,
    "spacingHuge": 64
  },
  "colorTokens": {
    "colorPrimary": "#F4AB34",
    "colorOnPrimary": "#333333",
    "colorPrimaryLight": "#FEFDE8",
    "colorOnPrimaryLight": "#333333",
    "colorPrimaryDark": "#EF8426",
    "colorOnPrimaryDark": "#333333",
    "colorSecondary": "#FF6B0B",
    "colorOnSecondary": "#000000",
    "colorSecondaryLight": "#FFF8E1",
    "colorOnSecondaryLight": "#000000",
    "colorSecondaryDark": "#FF5808",
    "colorOnSecondaryDark": "#000000",
    "colorBackground": "#FAFAFA",
    "colorOnBackground": "#333333",
    "colorSurface": "#FFFFFF",
    "colorOnSurface": "#333333",
    "colorHighlight": "#000000",
    "colorHighEmphasis": "#333333",
    "colorMediumEmphasis": "#777777",
    "colorLowEmphasis": "#BBBBBB",
    "colorLink": "#227BBD",
    "colorOnLink": "#FFFFFF",
    "colorSuccess": "#569A32",
    "colorOnSuccess": "#FFFFFF",
    "colorWarning": "#FCC433",
    "colorOnWarning": "#333333",
    "colorAlert": "#E74627",
    "colorOnAlert": "#FFFFFF"
  },
  "elevation": {
    "1": {
      "elevation": 1,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 1, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 1
    },
    "2": {
      "elevation": 2,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 2, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 2
    },
    "3": {
      "elevation": 3,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 3, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 4
    },
    "4": {
      "elevation": 4,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 4, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 5
    },
    "5": {
      "elevation": 5,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 6, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 10
    },
    "6": {
      "elevation": 6,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 8, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 10
    },
    "7": {
      "elevation": 7,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 9, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 12
    },
    "8": {
      "elevation": 8,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 12, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 17
    },
    "9": {
      "elevation": 9,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 16, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 24
    },
    "10": {
      "elevation": 10,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 24, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 38
    }
  }
}
