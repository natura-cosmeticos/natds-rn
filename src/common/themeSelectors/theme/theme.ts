import { tokens as styleTokens } from '@naturacosmeticos/natds-styles';
import { IColorThemeTokens } from '@naturacosmeticos/natds-styles/dist/tokens/colors/themeBound/IColorThemeTokens';
import { IElevation } from '@naturacosmeticos/natds-styles/dist/tokens/elevation/IElevation';

type Tokens = typeof styleTokens;
type TokensWithoutColor = Omit<Tokens, 'colors'>
type TokensWithoutColorTokens = Omit<TokensWithoutColor, 'colorTokens'>
type TokensWithoutElevation = Omit<TokensWithoutColorTokens, 'elevation'>

interface ElevationProps {
  elevation: number
  shadowColor: string
  shadowOffset: { height: number, width: number }
  shadowOpacity: number
  shadowRadius: number
}

type Elevation = Record<keyof IElevation, ElevationProps>


export type Theme = TokensWithoutElevation
  & {
    colorTokens: IColorThemeTokens,
    elevation: Elevation | {}
  }

type ThemeMode = 'dark' | 'light';

const parseElevation = (elevation: string) => {
  const regEx = /(\d+(\.\d+)?)/g;
  const matches = elevation.match(regEx);

  if (matches) {
    const [width, height, blurRadius, , r, g, b, alpha] = matches;

    const toUnit = (prop: string) => parseInt(prop, 0);

    return {
      elevation: 2,
      shadowColor: `rgb(${r}, ${g}, ${b})`,
      shadowOffset: { height: toUnit(height), width: toUnit(width) },
      shadowOpacity: parseFloat(alpha),
      shadowRadius: toUnit(blurRadius),
    };
  }

  return {};
};

const buidElevation = (elevation: IElevation) => {
  const nativeElevation = {};

  Object
    .keys(elevation)
    .forEach((key) => { nativeElevation[key] = parseElevation(elevation[key]); });

  return nativeElevation;
};

export const buildTheme = (tokens: Tokens, brand: string, mode: ThemeMode) => {
  const colorThemeName = `${brand}${mode === 'light' ? 'Light' : 'Dark'}ColorTokens`;
  const localTokens = { ...tokens };

  delete localTokens.colors;

  const theme: Theme = {
    ...localTokens,
    colorTokens: localTokens.colorTokens[colorThemeName],
    elevation: buidElevation(localTokens.elevation),
  };

  return theme;
};

interface ThemeSelector {
  (theme: Theme): string | undefined
}

export const getTheme = (selector: ThemeSelector) => (
  { theme }: { theme: Theme },
) => selector(theme);
