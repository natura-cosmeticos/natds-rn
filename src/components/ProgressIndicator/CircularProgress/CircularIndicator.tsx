import React from 'react';
import { Animated } from 'react-native';
import { withTheme } from 'styled-components/native';

import { Theme } from '../../../common/themeSelectors';
import {
  getLayerStyles, getLineLayerStyles, getContainerStyles, getLoopLayerStyles, getRootStyles,
} from './CircularProgress.styles';

export interface CircularProgressIndicatorProps {
  size: number;
  color: string;
  timer: Animated.Value;
  theme: Theme;
}

const CircularIndicatorComponent = ({
  size, color, timer, theme,
}: CircularProgressIndicatorProps) => {
  const minCircularRange = '45deg';
  const maxCircularRange = '765deg';

  const rootStyles = getRootStyles(size);
  const loopLayerStyles = getLoopLayerStyles();
  const layerStyle = getLayerStyles(size, timer.interpolate({
    inputRange: [0, 1],
    outputRange: [
      minCircularRange,
      maxCircularRange,
    ],
  }));
  const containerStyle = getContainerStyles(size);
  const lineStyle = getLineLayerStyles(color, theme, size);

  return (
    <Animated.View
      style={rootStyles}
      collapsable={false}
    >
      <Animated.View style={loopLayerStyles}>
        <Animated.View style={layerStyle}>
          <Animated.View
            style={containerStyle}
            collapsable={false}
          >
            <Animated.View style={containerStyle} collapsable={false}>
              <Animated.View style={lineStyle} />
            </Animated.View>
          </Animated.View>
        </Animated.View>
      </Animated.View>
    </Animated.View>
  );
};

const CircularIndicator = withTheme(CircularIndicatorComponent);

export { CircularIndicator };
