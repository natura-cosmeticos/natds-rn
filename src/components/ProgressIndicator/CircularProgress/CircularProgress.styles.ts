import {
  StyleSheet,
} from 'react-native';
import styled from 'styled-components/native';

import { ProgressIndicatorColors } from '../ProgressIndicator';
import { Theme, getColorPrimary, getColorSecondary } from '../../../common/themeSelectors';

export const Layer = styled.View<{
  size: number,
}>(({ size }) => ({
  height: size,
  width: size,
}));

export const Line = styled.View<{
  color: ProgressIndicatorColors;
  theme: Theme;
  size: number;
}>(({ color, theme, size }) => ({
  borderColor: color === 'secondary' ? getColorSecondary(theme) : getColorPrimary(theme),
  borderRadius: size / 2,
  borderWidth: size / 10,
  height: size,
  width: size,
}));

export const Container = styled.View(({ size }: { size: number }) => ({
  height: size / 2,
  overflow: 'hidden',
  width: size,
}));

export const Loop = styled.View(() => ({
  ...StyleSheet.absoluteFillObject,
  alignItems: 'center',
  justifyContent: 'center',
}));
