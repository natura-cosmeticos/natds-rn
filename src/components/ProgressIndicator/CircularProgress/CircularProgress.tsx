import React, { useEffect, useRef } from 'react';
import {
  Animated,
  Easing,
  Platform,
} from 'react-native';

import { ProgressIndicatorColors } from '../ProgressIndicator';
import {
  Loop,
  Layer,
  Container,
  Line,
} from './CircularProgress.styles';

export interface CircularProgressProps {
  size: number;
  color: ProgressIndicatorColors;
}

export const CircularProgress = ({
  size,
  color,
}: CircularProgressProps) => {
  /**
   * Duration specify how much the circle will take to make a 360deg loop around itself,
   * decrease it will speed up the animation speed and increase will slow the animation speed
   * The default speed is 1 second per loop
   */
  const duration = 1000;
  /**
   * This animation/Animated.timing, is responsible for looping the border around the view.
   */
  const timer = useRef(new Animated.Value(0)).current;
  const rotation = Animated.timing(timer, {
    duration,
    easing: Easing.linear,
    isInteraction: false,
    toValue: 1,
    useNativeDriver: Platform.OS !== 'web',
  });
  /**
 * The rotate animation will take from 0deg to 360deg to make a full loop around itself
 */
  const fullCircularRange = 360;
  const minCircularRange = '0deg';
  const maxCircularRange = `${fullCircularRange}deg`;
  const layerStyle = {
    transform: [
      {
        rotate: timer.interpolate({
          inputRange: [0, 1],
          outputRange: [
            minCircularRange,
            maxCircularRange,
          ],
        }),
      },
    ],
  };

  /**
   * Loop rotation animation continuously,
   * each time it reaches the end, it resets and begins again from the start.
   */
  function startRotation(): void {
    timer.setValue(0);
    Animated.loop(rotation).start();
  }

  /**
   * Reset the timer and loop the animation again on each update
   */
  useEffect(() => {
    startRotation();
  }, []);

  return (
    <Layer as={Animated.View} size={size}>
      <Loop as={Animated.View}>
        <Layer
          as={Animated.View}
          size={size}
          style={layerStyle}
        >
          <Container as={Animated.View} size={size}>
            <Line
              as={Animated.View}
              color={color}
              size={size}
            />
          </Container>
        </Layer>
      </Loop>
    </Layer>
  );
};
