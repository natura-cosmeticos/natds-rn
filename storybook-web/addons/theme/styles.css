.theme {
  font-weight: 400;
  line-height: 1.5;
  max-width: 800px;
  font-family: Roboto, Arial, sans-serif;
  color: #555;
  margin: 30px;
}

.theme__item {
  max-width: 600px;
  border-bottom: rgba(0,0,0,.1) 1px solid;
  display: flex;
  align-items: center;
  padding: 0 0 10px;
  margin: 0 0 10px;
}

.theme__name {
  flex: 2 0 auto;
  font-size: 18px;
  text-transform: capitalize;
  padding: 0;
}

.theme__button {
  background: rgba(0,0,0,.3);
  border: 0;
  border-radius: 4px;
  font-size: 1rem;
  color: #fff;
  text-transform: capitalize;
  cursor: pointer;
  padding: 7px 14px;
  margin: 0 10px 0 0;
}

.theme__button:hover {
  background: rgba(0,0,0,.2);
}

.theme__button:active {
  background: rgba(0,0,0,.25);
}

.theme__button--selected,
.theme__button--selected:hover {
  background: rgba(0,0,0,.6);
  cursor: default;
}
