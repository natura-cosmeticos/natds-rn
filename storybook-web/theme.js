/* eslint-disable sort-keys */
import { create } from '@storybook/theming/create';
import themes from '@naturacosmeticos/natds-themes/react-native';

const buildTheme = (theme, mode) => create({
  base: mode,
  colorPrimary: theme.color.primary,
  colorSecondary: theme.color.secondary,

  // UI
  appBg: theme.color.surface,
  appContentBg: theme.color.background,

  // Text
  textColor: theme.color.highEmphasis,
  textInverseColor: theme.color.lowEmphasis,

  // Toolbar
  barBg: theme.color.surface,
  barTextColor: theme.color.highEmphasis,

  // Brand
  brandTitle: 'Natura Design System',
});

export const dark = buildTheme(themes.natura.dark, 'dark');
export const light = buildTheme(themes.natura.light, 'light');
