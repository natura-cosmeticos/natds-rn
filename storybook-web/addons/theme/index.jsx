import React, { useEffect, Fragment } from 'react';
import { useAddonState, Consumer } from '@storybook/api';
import { TabButton, WithTooltip, TooltipLinkList } from '@storybook/components';
import {
  PANEL_ID, CHANGE, PARAM_KEY, getStoryBookTheme,
} from './shared';
import { light, dark } from '../../theme';

import './styles.css';

const DEFAULT_THEME = {
  name: 'natura',
  type: 'light',
};

const buildThemeProps = mode => ({ ...DEFAULT_THEME, type: mode });

function parseTheme(themes) {
  return Object.entries(themes).reduce((accum, [name, variants]) => {
    Object.entries(variants).forEach(([type]) => {
      accum.push({ name, type });
    });

    return accum;
  }, []);
}

const createBackgroundSelectorItem = ({ name, type }, onChange, selectedItem) => ({
  active: selectedItem.name === name && selectedItem.type === type,
  id: `${name} - ${type}`,
  onClick: () => onChange({ name, type }),
  title: `${name} - ${type}`,
});

const getDisplayedItems = (list, onChange, selectedItem) => {
  if (!list.length) {
    return [];
  }

  return [
    ...list.map(params => createBackgroundSelectorItem(params, onChange, selectedItem)),
  ];
};

const mapper = ({ api, state }) => {
  const story = state.storiesHash[state.storyId];
  const { themes = {} } = (story && api.getParameters(story.id, PARAM_KEY)) || {};

  return { items: parseTheme(themes) };
};

export default function Theme({ channel, api }) {
  const apiTheme = getStoryBookTheme();
  const initialTheme = apiTheme ? buildThemeProps(apiTheme) : DEFAULT_THEME;
  const [currentTheme, changeTheme] = useAddonState(PANEL_ID, initialTheme);
  const { disabled } = api.getCurrentParameter(PARAM_KEY) || {};

  const handleChange = (params) => {
    changeTheme(params);
    api.setOptions({
      theme: params.type === 'light' ? light : dark,
    });
    const event = new Event('CHANGE_THEME', { detail: params.type });

    window.dispatchEvent(event);
    channel.emit(CHANGE, params);
  };

  const renderTooltip = items => ({ onHide }) => (
    <TooltipLinkList
      links={getDisplayedItems(items, (index) => {
        handleChange(index);
        onHide();
      }, currentTheme)}
    />
  );

  const eventListener = event => handleChange(buildThemeProps(event.matches ? 'dark' : 'light'));

  useEffect(() => {
    handleChange(initialTheme);

    const matchMediaEnvent = window.matchMedia('(prefers-color-scheme: dark)');

    matchMediaEnvent.addEventListener('change', eventListener);

    return () => {
      matchMediaEnvent.removeEventListener('change', eventListener);
    };
  }, []);

  if (disabled) return null;

  return (
    <Consumer filter={mapper}>
      {({ items }) => (items.length ? (
          <Fragment>
            <WithTooltip
              placement="top"
              trigger="click"
              closeOnClick
              tooltip={renderTooltip(items)}
            >
              <TabButton
                key="Theme"
                title="Change the Theme">
                {currentTheme.name} - {currentTheme.type}
              </TabButton>
            </WithTooltip>
          </Fragment>
      ) : null)}
    </Consumer>
  );
}
