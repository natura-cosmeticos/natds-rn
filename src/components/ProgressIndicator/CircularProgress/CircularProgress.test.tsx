import React from 'react';

import { ThemeProvider } from 'styled-components/native';
import renderer from 'react-test-renderer';

import theme from '../../../common/themeSelectors/theme/mock-theme.json';
import { CircularProgress, CircularProgressProps } from './CircularProgress';

const defaultProps = {
  color: 'primary',
  size: 24,
  type: 'indeterminate',
  variant: 'circular',
} as Omit<CircularProgressProps, 'theme'>;

const renderCircularProgress = (props?: Omit<CircularProgressProps, 'theme'>) => (renderer.create(
  <ThemeProvider theme={theme}>
    <CircularProgress
      {...defaultProps}
      {...props}
    />
  </ThemeProvider>,
));

describe('CircularProgress component', () => {
  it('Should render circular progress with default props correctly', () => {
    const circularProgress = renderCircularProgress().toJSON();

    expect(circularProgress).toMatchSnapshot();
  });

  it('Should render circular progress with color primary correctly', () => {
    const circularProgress = renderCircularProgress({ ...defaultProps, color: 'primary' }).toJSON();

    expect(circularProgress).toMatchSnapshot();
  });

  it('Should render circular progress with color primary secondary', () => {
    const circularProgress = renderCircularProgress({ ...defaultProps, color: 'secondary' }).toJSON();

    expect(circularProgress).toMatchSnapshot();
  });
});
