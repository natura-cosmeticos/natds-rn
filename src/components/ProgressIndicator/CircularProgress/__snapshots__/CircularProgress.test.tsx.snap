// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularProgress component Should render circular progress with color primary correctly 1`] = `
<View
  size={24}
  style={
    Object {
      "height": 24,
      "width": 24,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      size={24}
      style={
        Object {
          "height": 24,
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
          "width": 24,
        }
      }
    >
      <View
        size={24}
        style={
          Object {
            "height": 12,
            "overflow": "hidden",
            "width": 24,
          }
        }
      >
        <View
          color="primary"
          size={24}
          style={
            Object {
              "borderColor": "#F4AB34",
              "borderRadius": 12,
              "borderWidth": 2.4,
              "height": 24,
              "width": 24,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`CircularProgress component Should render circular progress with color primary secondary 1`] = `
<View
  size={24}
  style={
    Object {
      "height": 24,
      "width": 24,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      size={24}
      style={
        Object {
          "height": 24,
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
          "width": 24,
        }
      }
    >
      <View
        size={24}
        style={
          Object {
            "height": 12,
            "overflow": "hidden",
            "width": 24,
          }
        }
      >
        <View
          color="secondary"
          size={24}
          style={
            Object {
              "borderColor": "#FF6B0B",
              "borderRadius": 12,
              "borderWidth": 2.4,
              "height": 24,
              "width": 24,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`CircularProgress component Should render circular progress with default props correctly 1`] = `
<View
  size={24}
  style={
    Object {
      "height": 24,
      "width": 24,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      size={24}
      style={
        Object {
          "height": 24,
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
          "width": 24,
        }
      }
    >
      <View
        size={24}
        style={
          Object {
            "height": 12,
            "overflow": "hidden",
            "width": 24,
          }
        }
      >
        <View
          color="primary"
          size={24}
          style={
            Object {
              "borderColor": "#F4AB34",
              "borderRadius": 12,
              "borderWidth": 2.4,
              "height": 24,
              "width": 24,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
